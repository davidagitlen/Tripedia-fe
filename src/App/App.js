import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Look, it's a React app!
        </p>
      </header>
    </div>
  );
}

export default App;