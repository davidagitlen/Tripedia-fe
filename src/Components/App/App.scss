body {
  min-height: 100vh;
  min-width: 100vw;
  margin: 0;

  .map-form__container {
    display: flex;
  }

  .map-container {
    height: 100%;
    width: 100%;
  }
  .app-is-loading {
    background-color: rgba(170, 164, 164, 0.75);
    position: fixed;
    padding: 0;
    margin: 0;
    top: 0;
    left: 0;
    z-index: 75;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    .trivia-box__container {
      width: 1200px;
      height: 600px;
      background-color: rgba(240, 240, 240, 0.98);
      border-radius: 8px;
      font-size: 34px;
      color: blue;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
      background: url("../../Images/loading-background.jpg");
      background-repeat: no-repeat;
      background-size: cover;

      .animation-container {
        width: 99.3%;
        height: 100px;
        // background: url('../../images/loading-background.jpg');
        // background-repeat: no-repeat;
        // background-size: fill;
        // background-position: 0%;
        .loading__icon {
          width: 100px;
          animation: drive 13s;
          animation-iteration-count: infinite;
          position: relative;
          z-index: 100;
        }
        .stop-sign {
          width: 50px;
          height: 50px;
          position: relative;
          left: 565px;
          bottom: 24px;
          z-index: 60;
        }

        .trees {
          width: 50px;
          height: 50px;
          position: relative;
          right: 21px;
          bottom: 24px;
        }

        .house {
          width: 50px;
          height: 50px;
          position: relative;
          right: 126px;
          bottom: 24px;
        }

        .hotel {
          width: 50px;
          height: 50px;
          position: relative;
          left: 386px;
          bottom: 24px;
        }

        .city {
          width: 75px;
          height: 75px;
          position: relative;
          left: 72px;
          bottom: 37px;
        }

        .city2 {
          width: 75px;
          height: 75px;
          position: relative;
          left: 104px;
          bottom: 37px;
        }

        .road {
          width: 100%;
          margin-top: -45px;
          height: 7px;
          box-shadow: 0px 0px 4px 4px darkgrey;
          background-color: darkgrey;
        }

        @keyframes drive {
          0% {
            transform: translateX(0px, 0px);
          }
          10% {
            transform: translateX(120px, -50px);
          }
          20% {
            transform: translateX(240px, -100px);
          }
          30% {
            transform: translateX(360px, -105px);
          }
          40% {
            transform: translateX(480px, -200px);
          }
          50% {
            transform: translateX(600px);
          }
          60% {
            transform: translateX(720px, 0px);
            // transform: translateY(-12px);
          }
          70% {
            transform: translateX(840px, -50px);
          }
          80% {
            transform: translateX(960px, -100px);
          }
          90% {
            transform: translateX(1080px, -105px);
            // transform: translateY(0);
          }
          100% {
            transform: translateX(1100px);
          }
        }
      }
    }
  }
}
