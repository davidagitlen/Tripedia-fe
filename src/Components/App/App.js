import React, { useMemo, useState, useContext } from "react";
import { Route } from "react-router-dom";
import Navigation from "../Navigation/Navigation";
import FormsContainer from "../FormsContainer/FormsContainer";
import "./App.scss";
import Map from "../Map/Map";
import LoginForm from "../LoginForm/LoginForm";
import CreateAccount from "../CreateAccount/CreateAccount";
import LoadingIcon from "../../Images/loading-icon.svg";
import stopSign from "../../Images/stop-sign.svg";
import trees from "../../Images/trees.svg";
import house from "../../Images/house.svg";
import hotel from "../../Images/hotel.svg";
import city from "../../Images/city.svg";
import { LoadingContext } from "../../Contexts/LoadingContext";
import { FormContext } from "../../Contexts/FormContext";
import { UserContext } from "../../Contexts/UserContext";
import { triviaFacts } from "../../util/trivia-facts";

export const App = () => {
  const [isLoadingState, setLoadingContext] = useState({ isLoading: false });
  const loadingState = useMemo(() => ({ isLoadingState, setLoadingContext }), [
    isLoadingState,
    setLoadingContext
  ]);
  const [formState, setFormState] = useState({
    origin: "",
    destination: "",
    tripSubmitted: false,
    attractions: {},
    accommodations: {},
    food: {},
    drinks: {},
    services: {},
    miscellaneous: {},
    selectedCategories: []
  });

  const currentForm = useMemo(() => ({ formState, setFormState }), [formState, setFormState]);
  const [user, userLogin] = useState({ email: '', name: '', id: ''});
  const loggedInUser = useMemo(() => ({ user, userLogin }), [ user, userLogin ]);
  const {isLoading} = isLoadingState;

  const selectTriviaFact = triviaArray => {
    let index = Math.floor(Math.random() * triviaArray.length);
    return triviaArray[index];
  };
  

  if (!isLoading) {
    return (
      <FormContext.Provider value={currentForm}>
        <UserContext.Provider value={loggedInUser}>
          <LoadingContext.Provider value={loadingState}>
            <div>
              <Route exact path="/login" render={() => <LoginForm />} />
              <Route
                exact
                path="/create_account"
                render={() => <CreateAccount />}
              />
              <Route
                exact
                path="/"
                render={() => (
                  <div className="nav_banner">
                    <Navigation />
                    <div className="map-form__container">
                      <FormsContainer />
                      <div className="map-container">
                        <Map
                          center={{ lat: 39.7392, lng: -104.9903 }}
                          zoom={11}
                        />
                      </div>
                    </div>
                  </div>
                )}
              />
            </div>
          </LoadingContext.Provider>
        </UserContext.Provider>
      </FormContext.Provider>
    );
  } else {
    return (
      <FormContext.Provider value={currentForm}>
        <UserContext.Provider value={loggedInUser}>
          <LoadingContext.Provider value={loadingState}>
            <div>
              <Route exact path="/login" render={() => <LoginForm />} />
              <Route
                exact
                path="/create_account"
                render={() => <CreateAccount />}
              />
              <Route
                exact
                path="/"
                render={() => (
                  <div className="nav_banner">
                    <Navigation />
                    <div className="map-form__container">
                      <FormsContainer />
                      <div className="map-container">
                        <Map
                          center={{ lat: 39.7392, lng: -104.9903 }}
                          zoom={11}
                        />
                      </div>
                    </div>
                  </div>
                )}
              />
            </div>
            <div className="app-is-loading">
              <div className="trivia-box__container">
                <p className="did-you-know__text">
                  While we prepare your trip, did you know:
                </p>
                <p>{selectTriviaFact(triviaFacts)}</p>
                <div className="animation-container">
                  <img
                    className="loading__icon"
                    alt="loading"
                    src={LoadingIcon}
                  ></img>
                  <img className="stop-sign" alt="loading" src={stopSign}></img>
                  <img className="house" alt="loading" src={house}></img>
                  <img className="trees" alt="loading" src={trees}></img>
                  <img className="trees" alt="loading" src={trees}></img>
                  <img className="trees" alt="loading" src={trees}></img>
                  <img className="trees" alt="loading" src={trees}></img>
                  <img className="trees" alt="loading" src={trees}></img>
                  <img className="city" alt="loading" src={city}></img>
                  <img className="city" alt="loading" src={city}></img>
                  <img className="city2" alt="loading" src={city}></img>
                  <img className="city2" alt="loading" src={city}></img>
                  <img className="hotel" alt="loading" src={hotel}></img>
                  <div className="road"></div>
                </div>
              </div>
            </div>
          </LoadingContext.Provider>
        </UserContext.Provider>
      </FormContext.Provider>
    );
  }
};

export default App;
