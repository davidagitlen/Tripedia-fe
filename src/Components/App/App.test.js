import React from 'react';
import ReactDOM from 'react-dom';
import { App }from './App';

describe('App', () => {
  it('should match the wrapper with data passed in', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
