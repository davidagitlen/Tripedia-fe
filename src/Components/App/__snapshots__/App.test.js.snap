// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the wrapper with data passed in 1`] = `
<ContextProvider
  value={
    Object {
      "formState": Object {
        "accommodations": Object {},
        "attractions": Object {},
        "destination": "",
        "destinationCity": "",
        "drinks": Object {},
        "food": Object {},
        "miscellaneous": Object {},
        "origin": "",
        "originCity": "",
        "selectedCategories": Array [],
        "services": Object {},
        "tripSubmitted": false,
        "trip_id": "",
        "waypoints": Array [],
      },
      "setFormState": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "user": Object {
          "email": "",
          "id": "",
          "name": "",
        },
        "userLogin": [Function],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "isLoadingState": Object {
            "isLoading": false,
            "loadingArray": Array [],
          },
          "setLoadingContext": [Function],
        }
      }
    >
      <div>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/create_account"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/map"
          render={[Function]}
        />
      </div>
    </ContextProvider>
  </ContextProvider>
</ContextProvider>
`;
